x-pack/legacy/plugins/canvas/public/components/router/router.js

```diff
@@ -57,12 +59,30 @@ export class Router extends React.PureComponent {
       // if this is the first load, execute the route
       if (firstLoad) {
         firstLoad = false;
+
+        // execute the route
         router
           .execute()
           .then(() => onLoad())
           .catch(err => onError(err));
       }
 
+      // dispatch actions based on the app state
+      const appState = getAppState();
+
+      if (appState.__fullscreen) {
+        this.props.setFullscreen(appState.__fullscreen);
+      }
+
+      if (appState.__refresh) {
+        this.props.setRefreshInterval(appState.__refresh * 1000);
```

Comment:
Once a refreshInterval is set, the `setRefreshInterval` action is dispatched on every route change.  I think these are probably only here to pull from the url on first load? Should these move into the firstLoad `if` block above?

As it stands, setting a refresh interval through the ui will fire the setRefreshInterval action, which updates the url, which triggers this pathChange which will fire the action again.

Reply:
Here is redux starting up for a url with values for `__refresh` and `__autoplay` for example.

![image](https://user-images.githubusercontent.com/515272/60464807-8a501700-9c1d-11e9-965a-7c21a32de509.png)

Reply:
Maybe it could move out of here entirely into the workpad routes.js file? Not sure, routing is always confusing.

Reply:
If I move it into `firstLoad`, changing the URL manually has no effect.

Reply:
I'm going to file a follow-up issue for this.  I don't want to add too much to the router so close to 7.3.  There may be a quick fix, I'll keep playing with it.

Reply:
👍 Yeah I don't think it's causing any issues as is